src/framework/uicomponents/view/abstractmenumodel.cpp

```diff
@@ -315,3 +319,30 @@ MenuItem& AbstractMenuModel::menu(MenuItemList& items, const QString& menuId)
     static MenuItem dummy;
     return dummy;
 }
+
+void AbstractMenuModel::updateShortcutsMenuItem(MenuItem* menuItem)
+{
+    auto screg = shortcutsRegister();
```

Comment:
This can just be inlined (i.e. no need for a variable).
Alternatively, if you want to call it less often, you could do the recursion a bit differently: instead of `MenuItem*`, let the recursive method take a `const std::vector<MenuItem*>&`

Reply:
Sorry, I wasn't aware of the mixed use of `QList` and `std::vector`. The conversion seems more cumbersome than the `shortcutsRegister` calls. Those are very lightweight anyway. So I've pushed another amendment myself.